superpeer/handler/MenuHandler.py

```diff
@@ -22,11 +29,77 @@ def serve(self, choice: str) -> None:
 		elif choice == "QUER":
 			pass
 
+		elif choice == "LISTSUPERPEERS":
+
+			shell.print_green('\nList of known peers:')
+
+			for count, friend in enumerate(LocalData.get_super_friends(), start=1):
+				friend_ip4 = LocalData.get_super_friend_ip4(friend)
+				friend_ip6 = LocalData.get_super_friend_ip6(friend)
+				friend_port = LocalData.get_super_friend_port(friend)
+				shell.print_blue(
+					f'{count}] {friend_ip4} {friend_ip6} {str(friend_port)}')
+
 		elif choice == "LISTPEERS":
-			pass
+
+			try:
+				conn = database.get_connection(db_file)
+				conn.row_factory = database.sqlite3.Row
+
+			except database.Error as e:
+				print(f'Error: {e}')
+				return "The server has encountered an error while trying to serve the request."
+
+			try:
+				peer_list = peer_repository.find_all(conn)
+
+				if peer_list is None:
+					shell.print_red('You do not know any peers.')
+					conn.close()
+
+				else:
+					for count, peer_row in enumerate(peer_list, start=1):
+						shell.print_blue(f'{count}]' + peer_row['ip'] + peer_row['port'] + '\n')
+
+			except database.Error as e:
+				conn.rollback()
+				conn.close()
+				print(f'Error: {e}')
+				return "The server has encountered an error while trying to serve the request."
 
 		elif choice == "LISTFILES":
-			pass
+			try:
+				conn = database.get_connection(db_file)
+				conn.row_factory = database.sqlite3.Row
+
+			except database.Error as e:
+				print(f'Error: {e}')
+				return "The server has encountered an error while trying to serve the request."
+
+			try:
+				files = file_repository.find_all(conn)
+
+				if files is None:
+					shell.print_red('You do not have any files.')
+					conn.close()
+
+				else:
+					for count, file_row in enumerate(files, start=1):
+						peer_list = peer_repository.get_peers_by_file(conn, file_row['file_md5'])
+
+						shell.print_green(f'{count}]' + file_row['file_md5'] + file_row['file_name'] + ':')
+						print('(')
+						for peer_row in peer_list:
+							peer_ip = peer_row['ip']
+							peer_port = peer_row['port']
+							print(f'{peer_ip} [{peer_port}]; ')
+						print(')\n')
+
+			except database.Error as e:
+				conn.rollback()
+				conn.close()
+				print(f'Error: {e}')
+				return "The server has encountered an error while trying to serve the request."
 
```

Comment:
Oggi io e @GiovanniFiorini stavamo ragionando sul fatto che il superpeer ha due sorgenti per i files in realtà:
1) Quella del database contente i file dei vicini (che giustamente viene pescata qua)
2) Quella dei propri files messi in condivisione, fatta con una lista presente in LocalData che ancora dobbiamo implementare.

Puoi fare conto che sia una lista chiamata ```shared_files``` composta da ```('nome_file', 'file_md5')```. @GiovanniFiorini correggimi se sbaglio/dimentico qualcosa.

Quindi niente @jhonrambo93, oltre a pescare i files dal DB, ricorda di pescarli anche da LocalData, magari implementatela pure con i metodi che ti servono, dopo che gio ci ha dato conferma sulla struttura della lista.

Reply:
Aspetto un feedback da @GiovanniFiorini